edgraph/server.go

```diff
@@ -698,10 +714,50 @@ func findVars(gmu *gql.Mutation) []string {
 	return varsList
 }
 
+// updateValMutations picks the val() from object and replaces it with its value
+// Assumption is that Subject should contain UID, whereas Object will VAL
+// If val(variable) exists in a query, but the values are not there for the variable,
+// it will delete the mutation itself, and stop it from execution
+func updateVALinMutation(gmu *gql.Mutation, varToVAL map[string]map[string]string) {
+	getNewVals := func(s string) (map[string]string, bool) {
+		if strings.HasPrefix(s, "val(") {
+			varName := s[4 : len(s)-1]
+			if vals, ok := varToVAL[varName]; ok {
+				return vals, true
+			}
+			return nil, true
+		}
+		return nil, false
+	}
+
+	gmuSet := gmu.Set[:0]
+	for _, nq := range gmu.Set {
+		newObs, exists := getNewVals(nq.ObjectId)
+		if !exists {
+			gmuSet = append(gmuSet, nq)
+			continue
+		}
+		val, ok := newObs[nq.Subject]
+		if !ok {
+			continue
+		}
+		nq.ObjectId = ""
+		nq.ObjectValue = &api.Value{
```

Comment:
I'm not sure whether it is accepted to write string value here. Does this work for other types as well?